remoulade/api/main.py

```diff
@@ -76,6 +86,25 @@ def get_actors():
     return {"result": [actor.as_dict() for actor in get_broker().actors.values()]}
 
 
+@app.route("/groups")
+def get_groups():
+    args = PageSchema().load(request.args.to_dict())
+    backend = remoulade.get_broker().get_state_backend()
+    groups = defaultdict(list)
+    states = (state for state in backend.get_states() if state.group_id)
+    if args.get("search_value"):
+        keys = ["message_id", "name", "actor_name", "args", "kwargs", "group_id"]
+        value = args["search_value"].lower()
+        states = [state for state in states if dict_has(state.as_dict(), keys, value)]  # type: ignore
+    for state in states:
+        groups[state.group_id].append(state.as_dict(exclude_keys=("args", "kwargs")))
+    groups = sorted(  # type: ignore
```

Comment:
make it more resilient

def key(group):
    try:
      return x["messages"][0]["enqueued_datetime"] or datetime.min
   except IndexError, KeyError:
     return datetime.min


also use reverse=True.